apptainer/petsc.def

```diff
@@ -90,6 +90,8 @@ From: {{ APPTAINER_FROM }}
 %files
     {{ MOOSE_DIR }}/scripts/configure_petsc.sh {{ ROOT_BUILD_DIR }}/scripts/configure_petsc.sh
     {{ MOOSE_DIR }}/scripts/update_and_rebuild_petsc.sh {{ ROOT_BUILD_DIR }}/scripts/update_and_rebuild_petsc.sh
+    {{ MOOSE_DIR }}/scripts/apple-silicon-hdf5-autogen.patch {{ ROOT_BUILD_DIR }}/scripts/apple-silicon-hdf5-autogen.patch
```

Comment:
I don't think this one is needed for linux, is it?

Reply:
no

Reply:
how do you optionally add files based on architecture?

Reply:
```bash
if [[ "${mpi}" == 'mpich' ]]; then
  do mpich related stuff
else
  do openmpi related stuff
fi
```

The 'keys' being set in `conda_build_config.yaml` become available as variables. So this section:
```yaml
mpi:
  - mpich
  - openmpi
```
controls what `mpi`'s value is at the time.